### PR TITLE
When duplicating a project, ask for the new name and if link with game should be kept

### PR DESCRIPTION
**Pull request by AlexandreSi**
image

### CODE REVIEW

File: newIDE/app/src/MainFrame/index.js [hunk 1]
```diff
@@ -2627,6 +2633,21 @@ const MainFrame = (props: Props) => {
           if (!canProjectBeSafelySavedAs) return;
         }
 
+        let originalProjectUuid = null;
+        if (newSaveAsOptions && newSaveAsOptions.generateNewProjectUuid) {
+          originalProjectUuid = currentProject.getProjectUuid();
+          currentProject.resetProjectUuid();
```
Comment (ClementPasteau) [inline newIDE/app/src/MainFrame/index.js:2639]: just double checking this will not affect the initial project? I assume not as we're not saving that project

File: newIDE/app/src/MainFrame/EditorContainers/HomePage/CreateSection/index.js [hunk 2]
```diff
@@ -314,7 +314,7 @@ const CreateSection = ({
     const isCurrentProjectOpened =
       !!project &&
       !!currentFileMetadata &&
-      fileMetadata.gameId === currentFileMetadata.gameId;
+      fileMetadata.fileIdentifier === currentFileMetadata.fileIdentifier;
```
Comment (4ian) [inline newIDE/app/src/MainFrame/EditorContainers/HomePage/CreateSection/index.js:317]: Good catch

File: newIDE/app/src/ProjectsStorage/CloudStorageProvider/CloudProjectWriter.js [hunk 3]
```diff
@@ -189,33 +190,40 @@ export const generateOnChooseSaveProjectAsLocation = ({
...
           closeDialog();
           resolve(null);
         }}
-        nameSuggestion={project.getName()}
-        onSave={(newName: string) => {
+        nameMaxLength={CLOUD_PROJECT_NAME_MAX_LENGTH}
+        nameSuggestion={
+          fileMetadata ? `${project.getName()} - Copy` : project.getName()
+        }
+        shouldAskForGameLinkRemoval={!!fileMetadata && !!fileMetadata.gameId}
```
Comment (ClementPasteau) [inline newIDE/app/src/ProjectsStorage/CloudStorageProvider/CloudProjectWriter.js:210]: maybe name this prop `shouldAskForNewProjectUuid` to be consistent with the props you use in the component?

File: newIDE/app/src/MainFrame/index.js (hunk 1 above)
Comment (AlexandreSi) [reply newIDE/app/src/MainFrame/index.js:2639]: In my tests everything went fine, but I'll check again

File: newIDE/app/src/ProjectsStorage/CloudStorageProvider/CloudProjectWriter.js [hunk 4]
```diff
@@ -243,7 +251,7 @@ export const generateOnSaveProjectAs = (
   }
   options.onStartSaving();
 
-  const gameId = saveAsLocation.gameId || project.getProjectUuid();
```
Comment (ClementPasteau) [inline newIDE/app/src/ProjectsStorage/CloudStorageProvider/CloudProjectWriter.js:246]: why this change? was this not used?

File: newIDE/app/src/ProjectsStorage/LocalFileStorageProvider/LocalProjectWriter.js [hunk 5]
```diff
@@ -180,16 +185,49 @@ export const onSaveProject = async (
...
+        }}
+      />
+    ));
+  });
+
+  if (!options) return { saveAsLocation: null, saveAsOptions: null }; // Save was cancelled.
+
+  let defaultPath = fileMetadata ? fileMetadata.fileIdentifier : '';
+  if (path && defaultPath) {
+    defaultPath = path.join(path.dirname(defaultPath), `${options.name}.json`);
```
Comment (ClementPasteau) [inline newIDE/app/src/ProjectsStorage/LocalFileStorageProvider/LocalProjectWriter.js:228]: any risk the `options.name` results in a filename that is not compatible with path? like with '/' in it?

File: newIDE/app/src/MainFrame/index.js (hunk 1 above)
Comment (AlexandreSi) [reply newIDE/app/src/MainFrame/index.js:2639]: I confirm it's okay

File: newIDE/app/src/ProjectsStorage/LocalFileStorageProvider/LocalProjectWriter.js (hunk 5 above)
Comment (AlexandreSi) [reply newIDE/app/src/ProjectsStorage/LocalFileStorageProvider/LocalProjectWriter.js:228]: Good remark, I'll add a check on the project name

File: newIDE/app/src/ProjectsStorage/CloudStorageProvider/CloudProjectWriter.js (hunk 4 above)
Comment (AlexandreSi) [reply newIDE/app/src/ProjectsStorage/CloudStorageProvider/CloudProjectWriter.js:246]: I saw it used only once, above in `generateOnChooseSaveProjectAsLocation`. I felt like it was not onChooseSaveProjectAsLocation role to set the gameId. So I removed it.

**Review by 4ian — APPROVED** [review]

**Review by 4ian — APPROVED** [review]
